datafusion/common/src/config.rs

```diff
@@ -251,6 +251,11 @@ config_namespace! {
         /// and sorted in a single RecordBatch rather than sorted in
         /// batches and merged.
         pub sort_in_place_threshold_bytes: usize, default = 1024 * 1024
+
+        /// Number of files to read in parallel when inferring schema and statistics
+        ///
+        /// Defaults to the number of CPU cores on the system
+        pub meta_fetch_concurrency: usize, default = num_cpus::get()
```

Comment:
I think it might be weird to use the number of cpu's as value here, as to benefit from it could be higher than the number of cpu's.
What about using the fixed value `32` here by default used previously?

Reply:
What is the final plan for this PR? Will we set the default to `32` ? Or shall we leave it at the number of cores?

Reply:
i will change it to `32`. this should be better. let me do that.

Reply:
@alamb Done.